Doc/library/pathlib.rst

```diff
@@ -1750,6 +1754,69 @@ Permissions and ownership
    symbolic link's mode is changed rather than its target's.
 
 
+Path information
+----------------
+
+.. class:: Info
+
+   Class that supports fetching and caching information about a local
+   filesystem path. An instance of this class is available from
+   :attr:`Path.info`; do not instantiate it directly.
+
+   This is an implementation of the :class:`pathlib.types.PathInfo` protocol
+   with an additional method specific to local paths (:meth:`~Info.stat`.)
+
+   .. versionadded:: next
+
+   .. method:: exists(*, follow_symlinks=True)
+
+      Return ``True`` if the path is an existing file or directory, or any
+      other kind of file; return ``False`` if the path doesn't exist.
+
+      If *follow_symlinks* is ``False``, return ``True`` for symlinks without
+      checking if their targets exist.
+
+   .. method:: is_dir(*, follow_symlinks=True)
+
+      Return ``True`` if the path is a directory, or a symbolic link pointing
+      to a directory; return ``False`` if the path is (or points to) any other
+      kind of file, or if it doesn't exist.
+
+      If *follow_symlinks* is ``False``, return ``True`` only if the path
+      is a directory (without following symlinks); return ``False`` if the
+      path is any other kind of file, or if it doesn't exist.
+
+   .. method:: is_file(*, follow_symlinks=True)
+
+      Return ``True`` if the path is a file, or a symbolic link pointing to
+      a file; return ``False`` if the path is (or points to) a directory or
+      other non-file, or if it doesn't exist.
+
+      If *follow_symlinks* is ``False``, return ``True`` only if the path
+      is a file (without following symlinks); return ``False`` if the path
+      is a directory or other non-file, or if it doesn't exist.
+
+   .. method:: is_symlink()
+
+      Return ``True`` if the path is a symbolic link (even if broken); return
+      ``False`` if the path is a directory or any kind of file, or if it
+      doesn't exist.
+
+   .. method:: stat(*, follow_symlinks=True)
+
+      Return an :class:`os.stat_result` object containing low-level
+      information about this path, like :func:`os.stat`. The result is cached
+      on the :class:`Info` object.
+
+      If *follow_symlinks* is ``False`` and this path is a symlink, return
+      information about the symlink rather than its target.
+
+      On Windows, if the path was generated by scanning a directory, then the
+      ``st_ino``, ``st_dev`` and ``st_nlink`` attributes of the
+      :class:`~os.stat_result` are always set to zero. Call :meth:`Path.stat`
+      to get these attributes.
```

Comment:
I'm worried about duplicating docs -- there's nothing preventing them from going out of sync.
What do you think about referencing `PathInfo` like this? (untested)
```rst
   .. method:: exists(*, follow_symlinks=True)
               is_dir(*, follow_symlinks=True)
               is_file(*, follow_symlinks=True)
               is_symlink()

      Return cached information, as in :class:`pathlib.types.PathInfo`.
```